src/chainstate/stacks/miner.rs

```diff
@@ -119,20 +167,27 @@ impl StacksBlockBuilder {
                 return Err(Error::InvalidStacksTransaction("Invalid transaction anchor mode for anchored data".to_string()));
             }
 
-            StacksChainState::process_transaction(clarity_tx, tx)?;
+            let (fee, receipt) = StacksChainState::process_transaction(clarity_tx, tx)?;
```

Comment:
The Clarity transaction will yield an error if the block limit is exceeded -- `CostBalanceExceeded` -- should we use that to determine when a block is full? I'm worried that having multiple ways of tracking the block limit will introduce some skew between what the miner thinks the used budget is versus what the budget is when the block is processed.

Reply:
Yes; good catch.  I had only tested this while deploying an expensive-to-analyze contract, which does _not_ error out with `CostBalanceExceeded`.  Will update this and add a test.

Reply:
Hmm, I think there's a bug in the `process_transaction()` method itself -- it's returning `Ok()` even if the runtime cost was exceeded while analyzing a smart contract.  Will fix that too.

Reply:
This has been fixed -- now, parse errors and analysis errors will be converted to cost errors, which are now caught here.